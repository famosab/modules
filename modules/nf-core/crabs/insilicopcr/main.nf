process CRABS_INSILICOPCR {
    tag "$meta.id"
    label 'process_medium'

    // WARN: Version information not provided by tool on CLI. Please update version string below when bumping container versions.
    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/crabs:0.1.1--pyhb7b1952_0':
        'biocontainers/crabs:0.1.1--pyhb7b1952_0' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("*.{fa,fasta}"), emit: fasta
    path "versions.yml"                  , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args    = task.ext.args ?: ''
    def prefix  = task.ext.prefix ?: "${meta.id}"
    def VERSION = '0.1.1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    crabs insilico_pcr \\
        --input $fasta \\
        --output ${prefix}.crabs.fa \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        crabs: $VERSION
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def VERSION = '0.1.1' // WARN: Version information not provided by tool on CLI. Please update this string when bumping container versions.
    """
    touch ${prefix}.fa

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        crabs: $VERSION
    END_VERSIONS
    """
}
