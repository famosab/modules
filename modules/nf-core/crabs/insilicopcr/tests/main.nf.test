nextflow_process {

    name "Test Process CRABS_INSILICOPCR"
    script "../main.nf"
    process "CRABS_INSILICOPCR"

    tag "modules"
    tag "modules_nfcore"
    tag "crabs"
    tag "crabs/insilicopcr"

    test("sarscov2 - fasta") {

        config "./nextflow.config"

        when {
            params {
                module_args = '--forward "GTCGGTAAAACTCGTGCCAGC" --reverse "CATAGTGGGGTATCTAATCCCAGTTTG"'
            }
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path +'genomics/sarscov2/genome/genome-ena.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out.versions,
                    file(process.out.fasta[0][1]).name,
                    ).match()
                }
            )
        }

    }

    test("sarscov2 - fasta - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path +'genomics/sarscov2/genome/genome-ena.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    process.out,
                    path(process.out.versions[0]).yaml
                    ).match() 
                }
            )
        }

    }

}
