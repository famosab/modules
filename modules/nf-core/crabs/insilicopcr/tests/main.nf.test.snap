{
    "sarscov2 - fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d334dbe811797a5944b67d096b125b7c"
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d334dbe811797a5944b67d096b125b7c"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T12:28:13.221768"
    },
    "sarscov2 - fasta": {
        "content": [
            [
                "versions.yml:md5,d334dbe811797a5944b67d096b125b7c"
            ],
            "test.crabs.fa"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-05T14:46:51.816788"
    }
}